src/stirling/source_connectors/socket_tracer/protocols/common/data_stream_buffer.h

```diff
@@ -92,10 +92,23 @@ class DataStreamBuffer {
 
   /**
    * Get timestamp recorded for the data at the specified position.
+   * If less than previous timestamp, timestamp will be adjusted to be monotonically increasing.
    * @param pos The logical position of the data.
    * @return The timestamp or error if the position does not contain valid data.
    */
-  StatusOr<uint64_t> GetTimestamp(size_t pos) const { return impl_->GetTimestamp(pos); }
+  StatusOr<uint64_t> GetTimestamp(size_t pos) {
+    StatusOr<uint64_t> timestamp_ns_status = impl_->GetTimestamp(pos);
+    if (!timestamp_ns_status.ok()) return timestamp_ns_status;
+    uint64_t current_timestamp_ns = timestamp_ns_status.ConsumeValueOrDie();
+    uint64_t prev_timestamp_ns = GetPrevTimestamp();
+    if (current_timestamp_ns < prev_timestamp_ns) {
+      LOG(WARNING) << "Detected non-monotonically increasing timestamp " << current_timestamp_ns
+                   << ". Adjusting to previous timestamp + 1: " << prev_timestamp_ns + 1;
+      return prev_timestamp_ns + 1;
```

Comment:
Probably we want:
```
if (current < prev) {
  current += 1;
}
SET_PREVIOUS(current);
return current;
```

Reply:
I believe that incrementing `current` by one may not be enough to make it greater than the previous timestamp. Did you mean `prev += 1`?

Reply:
I've adopted the structure you outlined, replacing `current += 1` with `current = prev + 1`